include/boost/iterator/new_iterator_tests.hpp

```diff
@@ -85,33 +76,35 @@ void readable_iterator_test(const Iterator i1, T v)
   BOOST_TEST(v1 == v);
   BOOST_TEST(v2 == v);
 
-# if !BOOST_WORKAROUND(__MWERKS__, <= 0x2407)
-  readable_iterator_traversal_test(i1, v, detail::is_postfix_incrementable<Iterator>());
+#if !BOOST_WORKAROUND(__MWERKS__, <= 0x2407)
+  readable_iterator_traversal_test(
+      i1, v,
+      std::integral_constant<
+          bool, detail::is_postfix_incrementable<Iterator>::value>{});
 
   // I think we don't really need this as it checks the same things as
   // the above code.
-  BOOST_STATIC_ASSERT(is_readable_iterator<Iterator>::value);
-# endif
+  static_assert(is_readable_iterator<Iterator>::value,
+                "Iterator must be readable.");
+#endif
 }
 
 template <class Iterator, class T>
-void writable_iterator_test(Iterator i, T v, T v2)
-{
+void writable_iterator_test(Iterator i, T v, T v2) {
   Iterator i2(i); // Copy Constructible
   *i2 = v;
 
-# if !BOOST_WORKAROUND(__MWERKS__, <= 0x2407)
+#if !BOOST_WORKAROUND(__MWERKS__, <= 0x2407)
   writable_iterator_traversal_test(
-      i, v2, mpl::and_<
-          detail::is_incrementable<Iterator>
-        , detail::is_postfix_incrementable<Iterator>
+      i, v2,
+      iterators::detail::conjunction<
+          std::integral_constant<bool, detail::is_incrementable<Iterator>::value>,
+          std::integral_constant<bool, detail::is_postfix_incrementable<Iterator>::value>
       >());
-# endif
+#endif
 }
 
-template <class Iterator>
-void swappable_iterator_test(Iterator i, Iterator j)
-{
+template <class Iterator> void swappable_iterator_test(Iterator i, Iterator j) {
```

Comment:
Please, keep `template` header and the function/class signature on separate lines. Here and everywhere else.